src/lib/es2017.object.d.ts

```diff
@@ -3,25 +3,14 @@ interface ObjectConstructor {
      * Returns an array of values of the enumerable properties of an object
      * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
      */
-    values<T>(o: { [s: string]: T } | { [n: number]: T }): T[];
+    values<T extends object>(o: T): T[keyof T][];

```

Comment:
do not use `keyof` here. same reasoning as https://github.com/Microsoft/TypeScript/issues/13989#issuecomment-279622620